### DISCUSSION TITLE
How to use `signIn` callback when using custom login page?

### DISCUSSION THREAD

**Discussion by joehdodd** [nextauthjs/next-auth#7102] [Help]
Hi All,

How can I make use of the `signIn` callback when I have a custom login page? I have the following code set up for the `signIn` callback...
```JavaScript
    callbacks: {
        async signIn({ user }) {
            const allowedEmails = process.env.EMAILS;
            if (allowedEmails.includes(user.email)) {
                return true;
            } else {
                console.log('redirect')
                return '/unauthorized';
            }
        },
    },
```
But the only time I'm redirected to `/unauthorized` is when I disable my custom login page. I checked the docs, but I'm not seeing anything that would help me figure out how to perform the redirect I need with a custom login page. I saw something in the docs about checking query params from the login page. When I check the `query` object, all I get back is the email address that was used, nothing that I could use to perform the redirect.

I've done a ton of Googling and searching on Stack Overflow for this. And scoured the docs. Thanks for any help!

**Comment by Manitej66** [comment]
From the docs: 
```
Redirects returned by this callback cancel the authentication flow. Only redirect to error pages that, 
for example, tell the user why they're not allowed to sign in.
```

if you want to redirect to a custom login page after incorrect login, you can do it like below:

```js
    const res = await signIn("credentials", {
      callbackUrl: "/",
      redirect: false, // use this to not trigger a page reload
      email: data.email.trim().toLowerCase(),
      password: data.password,
    });
    if (res.error) {
      router.replace("/custom-login")
    }
```
